packages/slice-machine/src/features/onboarding/helpers.tsx

```diff
@@ -0,0 +1,165 @@
+import { Text } from "@prismicio/editor-ui";
+import { Dispatch, SetStateAction, useEffect, useRef } from "react";
+
+import { useLocalStorageItem } from "@/hooks/useLocalStorageItem";
+import { useRepositoryInformation } from "@/hooks/useRepositoryInformation";
+
+export type OnboardingStepType =
+  | "createPage"
+  | "codePage"
+  | "addSlice"
+  | "writeContent"
+  | "pushModels";
+
+export type OnboardingStep = {
+  id: OnboardingStepType;
+  title: string;
+  description: string;
+};
+
+type OnboardingStepContentDefinition = {
+  title?: string;
+  content: string | (() => JSX.Element);
+  videoUrl: string;
+};
+
+export type OnboardingStepContent = Record<
+  OnboardingStepType,
+  OnboardingStepContentDefinition
+>;
+
+export type OnboardingStepStatuses = Record<OnboardingStepType, boolean>;
+
+const steps: OnboardingStep[] = [
```

Comment:
💡 #idea: there is quite a lot of things going on in this file, maybe this piece of data could be moved to other file e.g. `OnboardingGuide.data.ts` or `OnboardingGuide.const.ts` or just 'constants.ts` ?

Reply:
There isn't a clear pattern for organizing these, and since I didn't see any file with that naming pattern, I left it as is to get some feedback in the review.

Reply:
sure, the naming is actually least important to me, the point was to just move content & config out of the code to make the file less heavy
I'm also curious what are the patterns for that here :)

Reply:
But I also like to see things separate, so I just committed a refactor 😄